contracts/IdentityManager.sol

```diff
@@ -0,0 +1,211 @@
+  function addOwner(address sender, Proxy identity, address newOwner)
+    onlyAuthorized
+    onlyOlderOwner(identity, sender)
+    rateLimited(identity, sender)
+  {
+    owners[identity][newOwner] = now - userTimeLock;
+    OwnerAdded(identity, newOwner, sender);
+  }
+
+  /// @dev Allows a recoveryKey to add a new owner with userTimeLock waiting time
+  function addOwnerFromRecovery(address sender, Proxy identity, address newOwner)
+    onlyAuthorized
+    onlyRecovery(identity, sender)
+    rateLimited(identity, sender)
+  {
+    if (owners[identity][newOwner] > 0) throw;
+    owners[identity][newOwner] = now;
+    OwnerAdded(identity, newOwner, sender);
+  }
+
+  /// @dev Allows an owner to remove another owner instantly
+  function removeOwner(address sender, Proxy identity, address owner)
+    onlyAuthorized
+    onlyOlderOwner(identity, sender)
+    rateLimited(identity, sender)
+  {
+    delete owners[identity][owner];
+    OwnerRemoved(identity, owner, sender);
+  }
+
+  /// @dev Allows an owner to change the recoveryKey instantly
+  function changeRecovery(address sender, Proxy identity, address recoveryKey)
+    onlyAuthorized
+    onlyOlderOwner(identity, sender)
+    rateLimited(identity, sender)
+    validRecovery(recoveryKey)
+  {
+    recoveryKeys[identity] = recoveryKey;
+    RecoveryChanged(identity, recoveryKey, sender);
+  }
+
+  /// @dev Allows an owner to begin process of transfering proxy to new IdentityManager
+  function initiateMigration(address sender, Proxy identity, address newIdManager)
+    onlyAuthorized
+    onlyOlderOwner(identity, sender)
+  {
+    migrations[identity].startTime = now;
+    migrations[identity].newAddress = newIdManager;
+    MigrationInitiated(identity, newIdManager, sender);
+  }
+
+  /// @dev Allows an owner to cancel the process of transfering proxy to new IdentityManager
+  function cancelMigration(address sender, Proxy identity) onlyAuthorized onlyOwner(identity, sender) {
+    MigrationCanceled(identity, migrations[identity].newAddress, sender);
+    delete migrations[identity];
+  }
+
+  /// @dev Allows an owner to finalize and completly transfer proxy to new IdentityManager
+  /// Note: before transfering to a new address, make sure this address is "ready to recieve" the proxy.
+  /// Not doing so risks the proxy becoming stuck.
+  function finalizeMigration(address sender, Proxy identity) onlyAuthorized onlyOlderOwner(identity, sender) {
+    if (migrations[identity].startTime > 0 && migrations[identity].startTime + adminTimeLock < now) {
+      address newIdManager = migrations[identity].newAddress;
+      delete migrations[identity];
+      identity.transfer(newIdManager);
+      MigrationFinalized(identity, newIdManager, sender);
+    }
+  }
+
+  //Checks that address a is the first input in msg.data.
+  //Has very minimal gas overhead.
+  function checkMessageData(address a) constant internal returns (bool t) {
+    if (msg.data.length < 36) return false;
```

Comment:
hm, all tests work for me without that line. The only reason I ask is because solidity-coverage says that that line isn't covered, and I can't figure out a way to test it. But I guess it's fine to leave it there.